pandas/tests/indexing/multiindex/test_multiindex.py

```diff
@@ -70,6 +68,26 @@ def test_multi_nan_indexing(self):
         )
         tm.assert_frame_equal(result, expected)
 
+    def test_exclusive_nat_column_indexing(self):
+        # GH 38025
+        # test multi indexing when one column exclusively contains NaT values
+        df = DataFrame(
+            {
+                "a": [pd.NaT, pd.NaT, pd.NaT, pd.NaT],
+                "b": ["C1", "C2", "C3", "C4"],
+                "c": [10, 15, np.nan, 20],
+            }
+        )
+        df = df.set_index(["a", "b"])
+        expected = DataFrame(
+            {
+                "a": [pd.NaT, pd.NaT, pd.NaT, pd.NaT],
+                "b": ["C1", "C2", "C3", "C4"],
+                "c": [10, 15, np.nan, 20],
+            }
```

Comment:
```suggestion
                "c": [10, 15, np.nan, 20]
            },
            index=[
                Index([pd.NaT, pd.NaT, pd.NaT, pd.NaT], name="a"),
                Index(["C1", "C2", "C3", "C4"], name="b"),
            ]            
```
This should correct it. Passes in 1.3.3 and throws an AssertionError in 1.1.3, where the bug was reported.

Reply:
Thank you for your review. I fixed the previous problem. Should I still apply your suggestion?

Reply:
Yeah, as far as I can tell it should still be broken after your fix as the expected DataFrame's index is not equal to df. I made the suggestion with the dropped columns in mind before seeing your fix though so you'd have to reverse it before applying the suggestion.

Reply:
But I'm asserting the values of those dataframes, not the objects. If the logic of the test is flawed, I'll change it.

Reply:
True, I was asserting the object in my mockup. However, when testing in 1.1.3 with drop=False, the following is the result:
![image](https://user-images.githubusercontent.com/28331593/134748493-4e7ff408-6ae3-4a1a-bd6d-52825f44f10d.png)
The reported bug occurs only in the index and the column remains NaT, so the test will still fail to check for the regression. I believe going with the suggestion and asserting the entire object is the correct way to go here.

Reply:
I got it. Thank you for your information.

Reply:
@Xnot the problem with your suggestion is:

```
df:
a   b   c    
NaT C1  10.0
    C2  15.0
    C3   NaN
    C4  20.0
expected:

 a  b   a   b     c             
NaT C1 NaT  C1  10.0
    C2 NaT  C2  15.0
    C3 NaT  C3   NaN
    C4 NaT  C4  20.0

```
setting the index like that in `expected` doesn't drop the columns as `set_index` does.
any suggestion?

Reply:
If you just include "a" and "b" in the index argument without also putting them in the first argument they don't need to be dropped:
```python
expected = DataFrame(
    {
        "c": [10, 15, np.nan, 20]
    },
    index=[
        Index([pd.NaT, pd.NaT, pd.NaT, pd.NaT], name="a"),
        Index(["C1", "C2", "C3", "C4"], name="b"),
    ]
)
```